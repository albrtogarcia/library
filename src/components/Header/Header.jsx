import Logo from '../../assets/logo.svg?react'
import { NavLink, Link } from 'react-router-dom'
import './Header.scss'
import { IconListVideo, IconFileQuestion, IconFileCheck, IconUser } from '../../assets/icon'

export const Header = () => (
  <header className="header">
    <Link to="/" className="header-brand">
      <Logo />
      <span className="header-tagline">Portal del Profesional</span>
    </Link>

    <div className="header-nav">
      <NavLink to="/adjudicaciones" className="nav-item">
        <IconListVideo />
        <span>Adjudicaciones</span>
      </NavLink>
      <NavLink to="/prefacturacion" className="nav-item">
        <IconFileQuestion />
        <span>Prefacturación</span>
      </NavLink>
      <NavLink to="/facturacion" className="nav-item">
        <IconFileCheck />
        <span>Facturación</span>
      </NavLink>
    </div>

    <button type="button" className="header-account">
      <div className="account-avatar">
        <IconUser />
      </div>
      <div className="account-content">
        <span className="account-name">Nombre Apellido</span>
        <span className="account-company">Agencia</span>
      </div>
    </button>
  </header>
)
