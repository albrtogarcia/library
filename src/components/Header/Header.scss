.header {
  display: flex;
  padding: 1rem 2rem;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid var(--border, #E2E8F0);
  background: var(--white, #FFF);

  &-brand {
    display: flex;
    align-items: center;
    gap: var(--gap-regular, 1rem);
    flex-shrink: 0;
  }

  &-tagline {
    color: var(--brand-brand-main, #000);
    font-size: 1.125rem;
    font-weight: 500;
  }

  &-nav {
    display: flex;
    align-items: flex-start;
    gap: var(--gap-regular, 1rem);

    .nav-item {
      display: flex;
      padding: 0.5rem 1.5rem;
      justify-content: center;
      align-items: center;
      gap: 0.5rem;
      border-radius: 2.25rem;
      text-decoration: none;
      color: var(--brand-brand-main, #000);
      border: 1px solid var(--border, #E2E8F0);
      background: var(--white, #FFF);
      transition: all 0.2s ease-in-out;

      &:hover {
        border-radius: 36px;
        border: 1px solid var(--border, #E2E8F0);
        background: var(--border, #E2E8F0);
      }

      &.active {
        border-radius: 2.25rem;
        color: var(--white, #FFF);
        border: 1px solid var(--brand-brand-main, #000);
        background: var(--brand-brand-main, #000);
      }
    }
  }

  &-account {
    border: 0;
    background-color: transparent;
    display: flex;
    align-items: center;
    flex-direction: row-reverse;
    gap: var(--gap-regular, 1rem);
    flex-shrink: 0;
  }
}

.account {

  &-content {
    display: flex;
    align-items: flex-end;
    flex-direction: column;
    gap: .25rem;
  }

  &-name {
    color: var(--slate-700, #334155);
    font-size: 0.875rem;
    font-weight: 500;
  }

  &-company {
    color: var(--slate-500, #64748B);
    font-size: 0.75rem;
    font-weight: 500;
  }

  &-avatar {
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 1.25rem;
    color: var(--slate-500, #64748B);
    background-color: var(--slate-200, #E2E8F0);
    display: flex;
    justify-content: center;
    align-items: center;
  }
}