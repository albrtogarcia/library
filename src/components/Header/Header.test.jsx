import { describe, expect, it } from 'vitest'
import { Header } from './Header'

describe('Header', () => {
  it('matches snapshot', () => {
    const element = Header()
    expect(element).toMatchSnapshot()
  })
})
