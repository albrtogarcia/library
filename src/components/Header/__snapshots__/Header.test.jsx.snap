// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches snapshot 1`] = `
<header
  className="header"
>
  <Link
    className="header-brand"
    to="/"
  >
    <SvgLogo />
    <span
      className="header-tagline"
    >
      Portal del Profesional
    </span>
  </Link>
  <div
    className="header-nav"
  >
    <NavLink
      className="nav-item"
      to="/adjudicaciones"
    >
      <SvgListVideo />
      <span>
        Adjudicaciones
      </span>
    </NavLink>
    <NavLink
      className="nav-item"
      to="/prefacturacion"
    >
      <SvgFileQuestion />
      <span>
        Prefacturación
      </span>
    </NavLink>
    <NavLink
      className="nav-item"
      to="/facturacion"
    >
      <SvgFileCheck />
      <span>
        Facturación
      </span>
    </NavLink>
  </div>
  <button
    className="header-account"
    type="button"
  >
    <div
      className="account-avatar"
    >
      <SvgUser />
    </div>
    <div
      className="account-content"
    >
      <span
        className="account-name"
      >
        Nombre Apellido
      </span>
      <span
        className="account-company"
      >
        Agencia
      </span>
    </div>
  </button>
</header>
`;
